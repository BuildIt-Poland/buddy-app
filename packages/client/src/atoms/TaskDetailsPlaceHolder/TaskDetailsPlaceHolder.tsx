import React from 'react';
import makeStyles from '@material-ui/core/styles/makeStyles';
import Box from '@material-ui/core/Box';
import Skeleton from '@material-ui/lab/Skeleton';

const useStyles = makeStyles(theme => ({
  title: {
    height: '6rem',
    width: '70%',
    margin: theme.spacing(1, 0),
  },
  description: {
    height: '4rem',
    margin: theme.spacing(1, 0),
  },
  checkbox: {
    width: '3rem',
    height: '3rem',
  },
  status: {
    width: '15rem',
    height: '3rem',
    borderRadius: '4rem',
  },
  wrapper: {
    marginBottom: theme.spacing(4),
    marginTop: theme.spacing(-2.5),
  },
}));

const TaskDetailsPlaceHolder: React.FC = () => {
  const classes = useStyles();

  const PlaceHolderSubHeader = () => (
    <Box className={classes.wrapper}>
      <Box display='flex' alignItems='center' justifyContent='space-between'>
        <Skeleton variant={'text'} className={classes.title} />
        <Skeleton variant='circle' className={classes.checkbox} />
      </Box>
      <Skeleton variant='circle' className={classes.status} />
    </Box>
  );

  const PlaceHolderTaskDescription = () => (
    <>
      <Skeleton variant={'text'} className={classes.description} />
      <Skeleton variant={'text'} className={classes.description} />
      <Skeleton variant={'text'} className={classes.description} />
      <Skeleton variant={'text'} className={classes.description} />
      <Skeleton variant={'text'} className={classes.description} />
    </>
  );

  return (
    <>
      <PlaceHolderSubHeader />
      <PlaceHolderTaskDescription />
    </>
  );
};

export default TaskDetailsPlaceHolder;
