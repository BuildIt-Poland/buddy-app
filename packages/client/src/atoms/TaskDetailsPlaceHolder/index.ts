import TaskDetailsPlaceHolder from './TaskDetailsPlaceHolder';

export default TaskDetailsPlaceHolder;
