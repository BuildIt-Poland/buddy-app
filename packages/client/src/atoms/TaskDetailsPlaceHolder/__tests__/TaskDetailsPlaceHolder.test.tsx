import React from 'react';
import { create } from 'react-test-renderer';

import TaskDetailsPlaceHolder from '../TaskDetailsPlaceHolder';

jest.mock('@material-ui/lab/Skeleton', () => 'Skeleton');

describe('Component - TaskDetailsPlaceHolder', () => {
  test('renders correctly', () => {
    const component = create(<TaskDetailsPlaceHolder />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
