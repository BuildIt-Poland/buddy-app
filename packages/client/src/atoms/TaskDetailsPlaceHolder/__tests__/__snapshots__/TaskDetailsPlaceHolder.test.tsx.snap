// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - TaskDetailsPlaceHolder renders correctly 1`] = `
Array [
  <Box
    className="makeStyles-wrapper-5"
  >
    <Box
      alignItems="center"
      display="flex"
      justifyContent="space-between"
    >
      <Skeleton
        className="makeStyles-title-1"
        variant="text"
      />
      <Skeleton
        className="makeStyles-checkbox-3"
        variant="circle"
      />
    </Box>
    <Skeleton
      className="makeStyles-status-4"
      variant="circle"
    />
  </Box>,
  <Skeleton
    className="makeStyles-description-2"
    variant="text"
  />,
  <Skeleton
    className="makeStyles-description-2"
    variant="text"
  />,
  <Skeleton
    className="makeStyles-description-2"
    variant="text"
  />,
  <Skeleton
    className="makeStyles-description-2"
    variant="text"
  />,
  <Skeleton
    className="makeStyles-description-2"
    variant="text"
  />,
]
`;
